packages/react/src/ErrorBoundary.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, ErrorInfo, ReactElement, ReactNode, isValidElement } from 'react';
+
+type RenderFallback = () => ReactElement;
+
+export type Props = {
+    children: ReactNode;
+    fallback?: ReactElement | RenderFallback;
+};
+
+export type State = {
+    hasError: boolean;
+    error?: Error;
+};
+
+export class ErrorBoundary extends Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            hasError: false,
+            error: undefined,
+        };
+    }
+
+    static getDerivedStateFromError(error: Error) {
+        return { hasError: true, error };
+    }
```

Comment:
is this used anywhere?

Reply:
This is a lifecycle method that gets invoked after an error has been thrown by a component within the error boundary. https://legacy.reactjs.org/docs/react-component.html#static-getderivedstatefromerror